src/IAPWS_WaterProperties.f90

```diff
@@ -0,0 +1,4258 @@
+              + tr*(f02 + z1*(f05 + z3*f11) &
+                + tr*(f03 + z3*f10 &
+                  + tr*z2*(f07 + z4*p2*f16))))
+    vest3 = vpt3n + vc
+    ! region 3g (40 MPa<p<100 MPa)
+  ELSEIF (ireg3 == 7) THEN
+    pr = p*(1.0_SRK/103.0_SRK) - 1.0_SRK
+    p2 = pr*pr
+    p4 = p2*p2
+    tr = t*(1.0_SRK/600.0_SRK) - 1.0_SRK
+    t2 = tr*tr
+    t3 = t2*tr
+    t4 = t2*t2
+    t6 = t3*t3
+    p2t2 = p2*t2
+    vpt3n = g01 + tr*(g02 + t2*(g03 + t3*g04)) &
+            + pr*(g05 + t2*g06 &
+              + pr*(tr*t4*g07 &
+                + pr*(tr*(g08 + t3*(g09 + t4*g10)) &
+                  + p2*(g11 &
+                    + p2*pr*(t6*g12 &
+                      + p2*t4*(t6*g13 &
+                        + p4*(g14 &
+                          + p2*t6*(t6*t6*g15 &
+                            + p4*p4*(g16 &
+                              + p2t2*(g17 &
+                                + p2t2*(g18 &
+                                  + p2t2*(g19 + p2t2*g20))))))))))))
+    vest3 = vpt3n
+  ENDIF
+ENDFUNCTION vest3
+!
+!-------------------------------------------------------------------------------
+!> Returns the specific volume of the phase.  Works for saturated liquid or
+!> vapor.
+!> @param P    The saturation pressure
+!> @param T    The saturation temperature
+!> @param DEST The density of the fluid [kg/m**3]
+!> @param EPS  TODO - what is EPS
+!>
+FUNCTION diter3(P, T, DEST, EPS)
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: DEST
+  REAL(SRK), INTENT(IN) :: EPS
+  REAL(SRK) :: diter3
+
+  REAL(SRK) :: d1v, d2v, d1, d2, x
+
+  d1v = dest*0.999_SRK
+  d2v = dest*1.001_SRK
+  IF (dest <= DCrit) THEN
+    d1 = d1v
+    IF (d2v <= DCrit) THEN
+      d2 = d2v
+    ELSE
+      d2 = DCrit
+    ENDIF
+  ELSE
+    d2 = d2v
+    IF (d1v >= DCrit) THEN
+      d1 = d1v
+    ELSE
+      d1 = DCrit
+    ENDIF
+  ENDIF
+
+  CALL WNPT3(d1, d2, p, t, eps, x)
+  DITER3 = x
+ENDFUNCTION diter3
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
+FUNCTION NULLP3N(D, T, P)
+  REAL(SRK), INTENT(IN) :: D
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK) :: NULLP3N
+
+  REAL(SRK) :: V
+
+  V = 1.0_SRK/D
+  NULLP3N = PVT3N(V, T) - P
+ENDFUNCTION nullp3n
+!
+!-------------------------------------------------------------------------------
+!TODO needs header
+SUBROUTINE WNPT3(XA, XB, P, T, EPS, X)
+  REAL(SRK), INTENT(IN) :: XA
+  REAL(SRK), INTENT(IN) :: XB
+  REAL(SRK), INTENT(IN) :: P
+  REAL(SRK), INTENT(IN) :: T
+  REAL(SRK), INTENT(IN) :: EPS
+  REAL(SRK), INTENT(OUT) :: X
+
+  REAL(SRK) :: X1, F1, F2, F3, X2, P1, P3, X3
+  INTEGER(SIK) :: irem, i
+
+  X1 = XA
+
+  F1 = nullp3n(X1, T, P)
+  X3 = XB
+  F3 = nullp3n(X3, T, P)
+
+  x = 0.0_SRK
+  DO I = 1, 40
+    IF (F1 /= F3) THEN
+      X = X1 + (X3 - X1)*F1/(F1 - F3)
+    ELSE
+      EXIT
+    ENDIF
+    IF (X < 0.0_SRK) X = (X1 + X3)/2.0_SRK
```

Comment:
This function and `wnph3` (and probably others) have a decent number of `/2.0_SRK` that could be changed to multiplication.  Might be good to try to search on any divisions like that... maybe regex `/[0-9]*\.[0-9]*_` or something like that to get divisions by reals and not divisions by variables...

Reply:
yeah.... thought about it, but its technically not wrong.  I can do these pretty quickly, at least 2.0.  Note that I'm a little fearful to use Constants module because pi is used as a variable to be consistent with the documentation.  (there are also some cases where PI is 3.14...  I will change /2.0  to *0.5, some for /4.0... maybe /8.0

Reply:
Fixed all `/2.0_SRK`'s... also there were several `*1.0_SRK`'s.  I didn't find anything else (no divide by 4, 5, 8, or 10.